core/services/eth/log_broadcaster.go

```diff
@@ -113,6 +117,25 @@ func (b *logBroadcaster) Start() {
 		b.updateLogCursor(currentHeight, 0)
 	}
 
+	go b.awaitInitialSubscribers()
+}
+
+func (b *logBroadcaster) awaitInitialSubscribers() {
+Outer:
+	for {
+		select {
+		case r := <-b.chAddListener:
+			b.onAddListener(r)
+
+		case <-b.DependentAwaiter.AwaitDependents():
+			break Outer
+
+		case <-b.chStop:
+			close(b.chDone)
+			return
+		}
+	}
+
 	go b.startResubscribeLoop()
 }
```

Comment:
```suggestion
func (b *logBroadcaster) awaitInitialSubscribers() {
	for {
		select {
		case r := <-b.chAddListener:
			b.onAddListener(r)
		case <-b.AwaitDependents():
			go b.startResubscribeLoop()
			return
		case <-b.chStop:
			close(b.chDone)
			return
		}
	}
}
```
How about this? I find it a bit more clear.